homeassistant/components/flux/config_flow.py

```diff
@@ -0,0 +1,268 @@
+    }
+)
+
+
+def default_settings():
+    """Return object with the default settings for the Flux integration."""
+    settings_dict = {}
+    settings_dict[CONF_START_CT] = DEFAULT_START_COLOR_TEMP_KELVIN
+    settings_dict[CONF_SUNSET_CT] = DEFAULT_SUNSET_COLOR_TEMP_KELVIN
+    settings_dict[CONF_STOP_CT] = DEFAULT_STOP_COLOR_TEMP_KELVIN
+    settings_dict[CONF_ADJUST_BRIGHTNESS] = True
+    settings_dict[CONF_MODE] = DEFAULT_MODE
+    settings_dict[CONF_INTERVAL] = {"seconds": 30}
+    settings_dict[ATTR_TRANSITION] = {"seconds": 30}
+    return settings_dict
+
+
+class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Flux."""
+
+    VERSION = 1
+
+    @staticmethod
+    @callback
+    def async_get_options_flow(config_entry: config_entries.ConfigEntry) -> OptionsFlow:
+        """Get the options flow for the Flux component."""
+        return OptionsFlow(config_entry)
+
+    async def async_step_user(self, user_input=None) -> FlowResult:
+        """Handle the initial step."""
+
+        if user_input is not None:
+            user_input.update(default_settings())
+            return self.async_create_entry(title=user_input[CONF_NAME], data=user_input)
+
+        return self.async_show_form(
+            step_id="user",
+            data_schema=MINIMAL_FLUX_SCHEMA,
+        )
+
+    async def async_step_import(self, yaml_config: ConfigType) -> FlowResult:
+        """Handle import from configuration.yaml."""
+        # start with the same default settings as in the UI
+        entry_data = default_settings()
+
+        # remove the old two very similar options
+        brightness = yaml_config.pop(CONF_BRIGHTNESS, False)
+        disable_brightness_adjust = yaml_config.pop(
+            CONF_DISABLE_BRIGHTNESS_ADJUST, False
+        )
+
+        # combine them into the "new" option
+        if brightness or disable_brightness_adjust:
+            entry_data[CONF_ADJUST_BRIGHTNESS] = False
+
+        entry_data[CONF_INTERVAL] = {"seconds": yaml_config.pop(CONF_INTERVAL)}
+        entry_data[ATTR_TRANSITION] = {"seconds": yaml_config.pop(ATTR_TRANSITION)}
+
+        if CONF_START_TIME in yaml_config:
+            entry_data[CONF_START_TIME] = str(yaml_config.pop(CONF_START_TIME))
+
+        if CONF_STOP_TIME in yaml_config:
+            entry_data[CONF_STOP_TIME] = str(yaml_config.pop(CONF_STOP_TIME))
+
+        yaml_config.pop("platform")
+
+        # apply the rest of the remaining options
+        entry_data.update(yaml_config)
+
+        self._async_abort_entries_match(entry_data)
+
+        return self.async_create_entry(title=entry_data[CONF_NAME], data=entry_data)
+
+
+class OptionsFlow(config_entries.OptionsFlow):
+    """Handle flux options."""
+
+    def __init__(self, config_entry: config_entries.ConfigEntry) -> None:
+        """Initialize the flux options flow."""
+        self._config_entry = config_entry
+
+    def reset_values_to_default(self, user_input):
```

Comment:
Why do we want to use the default again?

Reply:
After using non-default values, there should be a way to use the default values again.
For example after configuring a sunset time, you decide you want to use the sunset time based on location and time of the year again.
With the yaml config this could be done simply by removing a key value pair from the yaml file.
With the GUI I could not find a simple way to "unset" a value, so I used this hack.

Most default values are static but the "default" values related to (sunset) time change during the year.

Reply:
Finally found out how to make the value clearable🥳

Reply:
@schelv What did you do to make the values clearable?

Reply:
I changed
```
vol.Optional(
                  CONF_START_TIME,
                  default=settings.get(CONF_START_TIME, UNDEFINED),
              ): TimeSelector(),
```
to 
```
vol.Optional(
                  CONF_START_TIME,
                  description={"suggested_value": settings.get(CONF_START_TIME)},
              ): TimeSelector(),
```

and proposed [this frontend merge request](https://github.com/home-assistant/frontend/pull/18590)

Reply:
I see. Does it mean this PR depends on a merge of that frontend PR? If so, please link to it in the PR description and add tag `awaiting-frontend`

Reply:
As far as I remember it is possible to clear the value without the frontend merge request. Not very easily though.